src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/CounterAggregator.cs

```diff
@@ -16,21 +25,57 @@ public CounterAggregator(bool isMonotonic)
 
         public override void Update(double value)
         {
-            lock (this)
+            // Get the deltas array.
+            PaddedDouble[] deltas = _deltas;
+            Debug.Assert(deltas.Length == Environment.ProcessorCount);
+
+            // Get the delta best associated with the current thread, preferring to use core ID rather than
+            // thread ID to reduce contention.
+            ref PaddedDouble delta = ref deltas[
+#if NETCOREAPP2_1_OR_GREATER
+                Thread.GetCurrentProcessorId()
+#else
+                Environment.CurrentManagedThreadId
```

Comment:
> Environment.CurrentManagedThreadId [](http://example.com/codeflow?start=15&length=35)

does this mean on netfx it is possible to be slower as we are not using the core Id?

Reply:
> > Environment.CurrentManagedThreadId
> 
> does this mean on netfx it is possible to be slower as we are not using the core Id?

Slower than with GetCurrentProcessorId, yes, as it's more likely for there to be contention.

Reply:
`Thread.CurrentThread.ManagedThreadId` should work fine on .NET Framework btw, it should use a jit opt that folds it to fast a helper call: https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAmAjAWAChMAGAAk1wDoAVACwCcBTRAEwEsA7AcwG4jSFXAFZ+hIgIDMFbGQDCZAN5EyqslzhkAsgAoN6gJRkAvAD4y9ZmypyArg2ac4llqypbEnRNyasXbAElWMTUyIgBfIA===